network/p2p/gossip/gossip.go

```diff
@@ -34,20 +36,42 @@ func NewGossiper[T any, U GossipableAny[T]](
 	log logging.Logger,
 	set Set[U],
 	client *p2p.Client,
-) *Gossiper[T, U] {
-	return &Gossiper[T, U]{
+	metrics prometheus.Registerer,
```

Comment:
Making sure people are fine w/ this as our signature. Avalanchego's context exposes a `MultiGatherer` so we'd be expected to register a `prometheus.NewRegistry` into the VM's context to use this.

Reply:
Yeah this function should be taking in a `prometheus.Registerer` 👍.